### PR TITLE
New syntax to add aliases to already existing definitions

### LINKED ISSUE #860: Add aliases to already existing definitions
I'd like to write a custom definitions file that adds new aliases to definitions from defaults_en.txt - as opposed to creating entirely new units.

However, I could not find a way to do it without scouting for the original definition line and writing it again whole.

For example, defaults_en.txt contains:
```
calorie = 4.184 * joule = cal = thermochemical_calorie = cal_th
```

If I want to add an alias, I have to write in my custom definitions file:
```
calorie = 4.184 * joule = cal = thermochemical_calorie = cal_th = CUSTOM_CAL
```
whereas I'd like to just write something like:
```
calorie = ... = CUSTOM_CAL
```

### PR DESCRIPTION
**Pull request by crusaderky**
Closes #860 
FYI @jthielen

### CODE REVIEW

File: pint/registry.py [hunk 1]
```diff
@@ -325,6 +330,13 @@ def _define_single_adder(self, key, value, unit_dict, casei_unit_dict):
         if casei_unit_dict is not None:
             casei_unit_dict[key.lower()].add(key)
 
+    def _define_alias(self, definition, unit_dict, casei_unit_dict):
+        unit = unit_dict[definition.name]
+        unit.add_aliases(*definition.aliases)
+        for alias in unit.aliases:
+            unit_dict[alias] = unit
+            casei_unit_dict[alias.lower()].add(alias)
```
Comment (crusaderky) [inline pint/registry.py]: I could not figure out how casei_unit_dict works - or find any tests for it. So I just copied what already is done for new units.

**Comment by crusaderky** [comment]
I can see how this same syntax could be potentially exploited in the future for localization; e.g. something like

```
@locale it
    @alias meter = metro
@end
```
```python
>>> ureg = pint.UnitRegistry(locale="it")
>>> ureg("1 meter"), ureg("1 metro")
1 metro, 1 metro
```
However that would require a lot of extra effort which is far beyond the scope of this PR.

**Comment by crusaderky** [comment]
Doesn't work with UnitRegistry.load_definitions() - working on a fix

**Comment by crusaderky** [comment]
Fixed

**Comment by crusaderky** [comment]
@hgrecco ready for review and merge

File: pint/registry.py (hunk 1 above)
Comment (hgrecco) [reply pint/registry.py]: That is fine

**Comment by hgrecco** [comment]
> I can see how this same syntax could be potentially exploited in the future for localization; e.g. something like
> 
> ```
> @locale it
>     @alias meter = metro
> @end
> ```
> 
> ```python
> >>> ureg = pint.UnitRegistry(locale="it")
> >>> ureg("1 meter"), ureg("1 metro")
> 1 metro, 1 metro
> ```
> 
> However that would require a lot of extra effort which is far beyond the scope of this PR.

It is a good idea, but I agree we need to move the dicussion to another PR. In particular, what worries me is that localization is not just aliasing but renaming

**Comment by hgrecco** [comment]
+1 for the docs in the PR.

**Comment by crusaderky** [comment]
Thanks!

> In particular, what worries me is that localization is not just aliasing but renaming

As somebody who had to endure working with localised Excel, I strongly advocate that localisation should impact exclusively the output of ``__repr__`` and ``__str__``.

**Pull request merged**